extensions/wikia/AdEngine/js/template/bfaa.js

```diff
@@ -1,48 +1,89 @@
-
-		if (doc.body.offsetWidth <= breakPointWidthNotSupported || position <= height) {
-			wrapper.classList.add('bfaa-pinned-nav');
-			nav.classList.add('bfaa-pinned');
-		} else {
-			wrapper.classList.remove('bfaa-pinned-nav');
-			nav.classList.remove('bfaa-pinned');
+		breakPointWidthNotSupported = 767, // SCSS property: $breakpoint-width-not-supported
+		desktopHandler,
+		mobileHandler,
+		nav,
+		page,
+		wrapper;
+
+	desktopHandler = {
+		updateNavBar: function (height) {
+			var position = win.pageYOffset;
+
+			if (doc.body.offsetWidth <= breakPointWidthNotSupported || position <= height) {
+				wrapper.classList.add('bfaa-pinned-nav');
+				nav.classList.add('bfaa-pinned');
+			} else {
+				wrapper.classList.remove('bfaa-pinned-nav');
+				nav.classList.remove('bfaa-pinned');
+			}
+		},
+
+		show: function (height, backgroundColor) {
+			nav.style.top = '';
+			page.classList.add('bfaa-template');
+			wrapper.style.background = backgroundColor;
+
+			this.updateNavBar(height);
+			doc.addEventListener('scroll', adHelper.throttle(function () {
+				this.updateNavBar(height);
+			}.bind(this), 100));
+
+			if (win.WikiaBar) {
+				win.WikiaBar.hideContainer();
+			}
 		}
-	}
+	};
 
-	function show(params) {
-		var backgroundColor = params.backgroundColor ? '#' + params.backgroundColor.replace('#', '') : '#000',
-			height = params.height || 0;
+	mobileHandler =  {
+		show: function (height, backgroundColor) {
+			win.Mercury.Modules.Ads.getInstance().setSiteHeadOffset(height);
+			page.style.paddingTop = height + 'px';
+			wrapper.style.background = backgroundColor;
+			page.classList.add('bfaa-template');
 
-		nav.style.top = '';
-		page.classList.add('bfaa-template');
-		wrapper.style.background = backgroundColor;
+			adContext.addCallback(function () {
+				page.classList.remove('bfaa-template');
+				wrapper.style.background = '';
+				page.style.paddingTop = '';
+				win.Mercury.Modules.Ads.getInstance().setSiteHeadOffset(0);
+			});
+		}
+	};
 
-		updateNavBar(height);
-		doc.addEventListener('scroll', adHelper.throttle(function () {
-			updateNavBar(height);
-		}, 100));
+	function show(params) {
+		var backgroundColor = params.backgroundColor ? '#' + params.backgroundColor.replace('#', '') : '#000',
+			handler,
+			height = params.height || 0,
+			skin = adContext.getContext().targeting.skin;
 
-		if (win.WikiaBar) {
-			win.WikiaBar.hideContainer();
+		switch (skin) {
+			case 'oasis':
+				nav = doc.getElementById('globalNavigation');
+				page = doc.getElementsByClassName('WikiaSiteWrapper')[0];
+				wrapper = doc.getElementById('WikiaTopAds');
+				handler = desktopHandler;
+				break;
+			case 'mercury':
+				page = doc.getElementsByClassName('application-wrapper')[0];
+				wrapper = doc.getElementsByClassName('mobile-top-leaderboard')[0];
+				handler = mobileHandler;
+				break;
+			default:
+				return log(['show', 'not supported skin'], 'info', logGroup);
 		}
 
+		handler.show(height, backgroundColor);
```

Comment:
Yup, it shouldn't be undefined and it's implemented in the same way like old modal = we have log for unsupported skin and it doesn't break anything.